tsdb/head.go

```diff
@@ -2048,6 +2048,31 @@ func GenerateTestHistograms(n int) (r []*histogram.Histogram) {
 	return r
 }
 
+func GenerateTestGaugeHistograms(n int) (r []*histogram.Histogram) {
+	for i := 0; i < n; i++ {
```

Comment:
`i` is supposed to be random here to get gauge histograms that have a counter reset in the context of counter histograms. See `GenerateTestGaugeFloatHistograms`.

Reply:
Ah yes, smart… and fixed.